deployments/helm/hephaestus/templates/controller/clusterrole.yaml

```diff
@@ -74,3 +74,11 @@ rules:
       - get
       - watch
       - patch
+  {{- if and (include "hephaestus.pspRequired" .) (include "hephaestus.istioWithoutCNI" .) }}
```

Comment:
why are both required?

Reply:
Good question, here's the answer:

You already know Istio w/o CNI means every Pod that is launched requires extra CAPs. and a root user. Hence, we need to do a little extra work whenever PSPs are enabled and configured to be _strict_.

The PSP API will be removed in k8s 1.25 so `include "hephaestus.pspRequired" .` macro will ensure that we don't render them whenever we start deploying to that version or greater. Without it, we'll get errors when trying to create resources for undefined object APIs and, in the case of RBAC objects, create superfluous rules that might confuse people

Reply:
I know why its needed with CNI, but given we have enablePSPs as a separate thing it should be an OR not an AND. PSPs are there for other things as well.
I agree the additional CAPs should be guarded by CNI but not the rest.

Reply:
changed the condition.